sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -267,28 +269,14 @@ void WinHttpTransport::CreateRequestHandle(std::unique_ptr<_detail::HandleManage
 void WinHttpTransport::Upload(std::unique_ptr<_detail::HandleManager>& handleManager)
 {
   auto streamBody = handleManager->m_request.GetBodyStream();
-  int64_t streamLength = streamBody->Length();
 
-  int64_t uploadChunkSize = handleManager->m_request.GetUploadChunkSize();
-  if (uploadChunkSize <= 0)
-  {
-    // use default size
-    if (streamLength < _detail::MaximumUploadChunkSize)
-    {
-      uploadChunkSize = streamLength;
-    }
-    else
-    {
-      uploadChunkSize = _detail::DefaultUploadChunkSize;
-    }
-  }
-
-  auto unique_buffer = std::make_unique<uint8_t[]>(static_cast<size_t>(uploadChunkSize));
+  auto unique_buffer
+      = std::make_unique<uint8_t[]>(static_cast<size_t>(_detail::DefaultUploadChunkSize));
```

Comment:
Only use `DefaultUploadChunkSize` if `streamLength` is > `MaximumUploadChunkSize`, otherwise use `MaximumUploadChunkSize`.

Reply:
why?
We just need the default now.  Since it won't be customizable anymore.

Reply:
We only want to split the buffer into 64k chunks if the entire buffer is >= 1 MB. When it's smaller, we just want to do it in one-shot.

Removing the `MaximumUploadChunkSize` is changing the logic here, unrelated to the customization of defaults. Please keep that logic as is, especially in this PR.

Reply:
Then why not setting the default to 1MB?  
Why do we prefer to handle 2 possible flows (we are not even testing it).  We can always just use 1MB,

Reply:
1) I don't think we should be discussing or changing implementation detail of a transport adapter as part of this change. Let's keep the change limited to what we actually decided. There is a buffering heuristic in place, which was done intentionally. Other changes should be done separately.
2) The purpose of the 64k vs 1 MB separation was to avoid creating a large 1MB vector when the stream is actually small. So the heuristic is, if the stream is less than 1 MB, use the actual stream size (don't create a 1 MB vector when the stream is 5 KB). If the stream is larger, use 64 KB chunks. I agree with you that it is reasonable to say, use 1 MB chunk rather than 64KB. for that (and I want to make this change after doing before/after perf comparison).

I expect this PR to keep the following code:
```C++
  // Consider using `MaximumUploadChunkSize` here, after some perf measurements
  int64_t uploadChunkSize = _detail::DefaultUploadChunkSize;
  if (streamLength < _detail::MaximumUploadChunkSize)
  {
    uploadChunkSize = streamLength;
  }
  auto unique_buffer = std::make_unique<uint8_t[]>(static_cast<size_t>(uploadChunkSize));
```

Reply:
fine.
Just have in mind that the mentioned heuristic can be the source of bugs.
I wouldn't expect any significant different from allocating 64k vs 1Mb.  The difference will be when it comes to copying the source to the allocated memory for uploading, we basically are expecting that
- One only 1MB allocate, copy and upload will be better than.
- One only 64k allocate and multiple 64k copy and upload operation.

But after copying 1Mb to the allocated memory, you will have the bottle neck on network because ssl will only upload 64k at a time, so the upload operation will do the iteration of read-and-upload until all is done

Reply:
> I wouldn't expect any significant different from allocating 64k vs 1Mb.

Is this allocating the array on the stack? Shouldn't we avoid unnecessary allocation of space that we don't need or use, even if there is now runtime perf cost of allocating a static array of 64 KB vs 1 MB?
`auto unique_buffer = std::make_unique<uint8_t[]>(static_cast<size_t>(uploadChunkSize));`


>  because ssl will only upload 64k at a time

Really?! With WinHTTP? Can you please share some documentation which mentions this limitation because this is news to me.